tests/test_cli.py::test_lazy_load_error randomly fails with PyPy3

After upgrading to 2.0.2, the tests sometimes (but not reliably) fail with pypy3.8:

```
============================================================== FAILURES ===============================================================
________________________________________________________ test_lazy_load_error _________________________________________________________
tests/test_cli.py:328: in test_lazy_load_error
    lazy._flush_bg_loading_exception()
E   Failed: DID NOT RAISE <class 'test_cli.test_lazy_load_error.<locals>.BadExc'>
======================================================= short test summary info =======================================================
FAILED tests/test_cli.py::test_lazy_load_error - Failed: DID NOT RAISE <class 'test_cli.test_lazy_load_error.<locals>.BadExc'>
=================================== 1 failed, 504 passed, 2 skipped, 2 xfailed, 1 xpassed in 10.40s ===================================
ERROR: InvocationError for command /tmp/flask/.tox/pypy3/bin/pytest -v --tb=short --basetemp=/tmp/flask/.tox/pypy3/tmp tests examples (exited with code 1)
_______________________________________________________________ summary _______________________________________________________________
ERROR:   pypy3: commands failed
```

I am able to replicate with every ~third invocation of:

```
tox -e pypy3
```

However, I was unable to reproduce when running just that one test. My totally random guess is that it might have something to do with PyPy's GC behavior.

Environment:

- Python version:
  ```
  Python 3.8.12 (7e233b73b92740ab7a20680a404ac100be1eb508, Sep 22 2021, 14:08:24)
  [PyPy 7.3.6 with GCC 11.2.0]
  ```
- Flask version: git 1cf43e08e3211e7ff21005dd778613e39b20a449 (happens since 2305b056c3a3161c92ea61e6fceb67262766bc40, i.e. adding the test)

--- a/tests/test_cli.py
+++ b/tests/test_cli.py
@@ -334,7 +334,9 @@ def test_lazy_load_error(monkeypatch):
     lazy = DispatchingApp(bad_load, use_eager_loading=False)
 
     with pytest.raises(BadExc):
-        lazy._flush_bg_loading_exception()
+        # reduce flakiness by waiting for the internal loading lock
+        with lazy._lock:
+            lazy._flush_bg_loading_exception()
 
 
 def test_with_appcontext(runner):

